test/blackbox/docker-compose.yaml

```diff
@@ -0,0 +1,103 @@
+services:
+
+  # The test_runner is the outside of the black box, it has a mounted copy of the full
+  # codebase for running tests etc.
+  test_runner:
+    image: php:8.2-alpine
+    environment:
+      - TEST_SUBJECT_BASE_URI=http://test_subject
+    working_dir: /workspace
+    volumes:
+      - ../../:/workspace
+    depends_on:
+      - test_subject
+    entrypoint: php
+    command:
+      - vendor/bin/phpunit
+      - --testsuite
+      - blackbox
+
+  # The test subject is the sample implementation we're testing
+  test_subject:
+    build:
+      context: ../../
+      dockerfile_inline: |
+        # First package the library in the current version, together with its dependencies
+        FROM composer:2 AS composer-builder
+        WORKDIR /workspace
+        COPY . /workspace
+        RUN  /workspace/test/blackbox/package-for-blackbox-tests.sh
+        
+        # Then build the packaged library into an HTTP server with our expected config and environment
+        # Note that we need to include composer to ensure that it will 
+        FROM php:8.2-apache
+        WORKDIR /var/www      
+        COPY --from=composer-builder /workspace/test/blackbox/implementation /var/www
+        COPY --from=composer:2 /usr/bin/composer /usr/bin/composer
+        RUN composer check-platform-reqs \
+            && a2enmod \
+              expires \
+              headers \
+              rewrite
+    configs:
+      - source: apache_config
+        target: /etc/apache2/sites-available/000-default.conf
+    ports:
+      - 80:80
+    volumes:
+      - ./implementation/htdocs/dynamic:/var/www/htdocs/dynamic
+    depends_on:
+      - test_subject_log_receiver
+    logging:
+      driver: syslog
+      options:
+        syslog-address: 'tcp://127.0.0.1:9002'
+        syslog-format: 'rfc3164'
+
+  test_subject_log_receiver:
+    # No real need to use the php image - it's just running netcat as a simple TCP > file dump - but saves another pull
+    image: php:8.2-alpine
```

Comment:
could just use `alpine:latest` it's layers will already be there unless you take on my comment above but it's only 3Mb if you want to be explicit it's got nothing to do with PHP